patches/minecraft/net/minecraft/entity/item/minecart/AbstractMinecartEntity.java.patch

```diff
@@ -17,6 +25,22 @@
  
     protected AbstractMinecartEntity(EntityType<?> p_i48538_1_, World p_i48538_2_) {
        super(p_i48538_1_, p_i48538_2_);
+@@ -104,6 +_,15 @@
+       this.field_70166_s = p_i48539_7_;
+    }
+ 
++   public net.minecraftforge.common.IMinecartCollisionHandler getCollisionHandler() {
++      return COLLISIONS;
++   }
++
++   // IMinecartCollisionHandler redirects here via comment. We ensure it exists and it works.
```

Comment:
This comment doesn't make much sense as it would seem "we ensure it exists and it works" is talking about the IMinecartCollisionHandler, but it doesn't seem that any verification actually happens here (in registerCollisionHandler)

Reply:
its' existence in this class means it works.
It currently does not work.
I agree the wording is terrible :P

Reply:
Note that also, the handler's default state is null, and thus null must remain a valid option for the setter, which means any checks we would reasonably need to do are done implicitly by the JVM when the setter is called.

Reply:
Ya I understand not checking anything here makes sense, my more point is that second part of the comment potentially shouldn't be there at all. The first part should make it clear enough why it exists/is unused but is needed

Reply:
True enough. I'll remove it. Give me one moment.